### ISSUE TITLE
Merging 2.0 branch into the master branch

### ISSUE THREAD

**Issue by ferdianjovan** [strands-project/soma#31]
Hi all,

Is it possible to merge 2.0 into the master branch?
In that way, we can re-release soma with its newer version.
I dont think anybody still uses the old version of soma.

**Comment by ferdianjovan** [comment]
@kunzel , @hkaraoguz , @PDuckworth , @jayyoung  
what do you think?

from my side, I have imported my work to use soma 2.0

**Comment by marc-hanheide** [comment]
I opened a PR for this #32 but it's not automatically merge-able, so apparently you did make modifications to both branches in parallel that need to be sorted out first.

**Comment by marc-hanheide** [comment]
A good first start would be to merge the `master` branch back into `2.0`?

**Comment by PDuckworth** [comment]
As an FYI, in Leeds we still use the original soma.

**Comment by marc-hanheide** [comment]
@hawesie, whats the plan to resolve this for tsc?

**Comment by hawesie** [comment]
@marc-hanheide quite a lot is running on Betty from source. We'll stick with that for TSC, but try to work through this at the same time.

@PDuckworth what are you using the original version for, and can you use it from source?

**Comment by marc-hanheide** [comment]
OK, @hawesie running from source is not a problem, but you still can't have two versions of the same package in parallel, so which SOMA is going to run? Has nothing to do with releasing it (though I _strongly_ advise releasing a version, in order to be able to build PR on dependent packages)

**Comment by hawesie** [comment]
@marc-hanheide agreed! (and AFAIK the Leeds dependency on 1.0 is not in the TSC system)

**Comment by ferdianjovan** [comment]
The skeleton tracker stuff from Leeds does not depend on soma at all. CMIIW

**Comment by marc-hanheide** [comment]
So, we agree to merge 2.0 into master and make that the default version also to release?

**Comment by kunzel** [comment]
As both soma versions are incompatible, the master branch should be simply replaced by the 2.0 branch (so, no merging is needed). However, before we do that, we should keep a copy of the outdated master branch for people who might still use it (maybe as 1.0).

BTW if people want to use the old soma version for their own work in the deployment they can use their own workspace and e.g. create ROIs in Mongo using the old format (as different collections are used). However, if people require the SAME regions and some of them are required to be in SOMA 2.0, then using the old version is not an option.

**Comment by marc-hanheide** [comment]
So, just to clarify: v2.0 _should_ indeed be the new master? So, we have to options:
1. Changing the release branch in bloom to v2.0, bump the major version an release.
2. Merging v2.0 into master manually, with any potential bugfix commits in the v1.0 preserved. It's a bit unfortunate those two diverged so much in the first place. The v1.0 will still be available as all releases are tagged anyway. So nothing needs to be done to preserve it but you may want to tag the last v1.0 commit again before merging. 

But if v2.0 is now the default then it either should be configured as default branch also in github and the Jenkins config changed to use that branch or it should be merged into master. Your call.

**Comment by marc-hanheide** [comment]
If you want to go for option 2 let me know as I can reconfigure things quickly, but I feel it would be the less clean option.

**Comment by marc-hanheide** [comment]
Hey, did you make progress towards this? People that need soma2 are knocking on my door as they get lots of failed pull requests, the likes of @bfalacerda for instance.

**Comment by hawesie** [comment]
My preference is for option 1as that is the widely used version now, with a v1.0 branch explicitly created (as not everyone knows to look at the tags).

**Comment by marc-hanheide** [comment]
I just correct my previous comment, as I meant I also prefer option 1 ;-)

**Comment by marc-hanheide** [comment]
So, who's in charge of this? @kunzel ?

**Comment by kunzel** [comment]
OK, I will take care about this next week after the ICRA workshop. I will also add a meta package for soma #23 and find a solution regarding #33 (I'd suggest that we separate the srv definitions out then @Pandoro OK?).

**Comment by hawesie** [comment]
I've been doing some of this. **NOTE TO ALL**: the branch 2.0 is deprecated. `indigo-devel` is the new release branch.
